activerecord/lib/active_record/railtie.rb

```diff
@@ -140,7 +140,17 @@ class Railtie < Rails::Railtie # :nodoc:
     initializer "active_record.define_attribute_methods" do |app|
       config.after_initialize do
         ActiveSupport.on_load(:active_record) do
-          descendants.each(&:define_attribute_methods) if app.config.eager_load
+          if app.config.eager_load
+            descendants.each do |model|
+              # These internal models override `table_exists?` to bypass the schema cache,
+              # so don't call `define_attribute_methods` on them to avoid the extra query.
+              next if model == ActiveRecord::SchemaMigration || model == ActiveRecord::InternalMetadata
```

Comment:
Don't know if it is worth, but we could add a `internal?` or `_internal?` if we don't want to pollute the `ActiveRecord::SchemaMigration` that returns `false` by default but return true for those two models.